inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +225,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
         else:
-            time.sleep(time_to_wait_before_reboot)
-        logger.debug("Rebooting to recover from failed SOTA...")
-        rebooter.reboot()
+            time.sleep(time_to_wait_before_reboot)            
+            if rebooter.is_reboot(self._device_reboot):    
+                logger.debug("Rebooting to recover from failed SOTA...")
+                rebooter.reboot()
```

Comment:
**Suggestion:** The time.sleep(time_to_wait_before_reboot) call is duplicated in both the recover and revert methods of the DebianBasedSnapshot class. This could be refactored into a private method to avoid code duplication and improve maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            self._wait_and_reboot_if_needed(rebooter, time_to_wait_before_reboot)

            # Add this private method to the DebianBasedSnapshot class
            def _wait_and_reboot_if_needed(self, rebooter: Rebooter, time_to_wait: int) -> None:
                time.sleep(time_to_wait)
                if rebooter.is_reboot(self._device_reboot):    
                    logger.debug("Rebooting to recover from failed SOTA...")
                    rebooter.reboot()
```